src/Microsoft.ML.Api/PredictionFunction.cs

```diff
@@ -7,30 +7,40 @@
 
 namespace Microsoft.ML.Runtime.Data
 {
+    public sealed class PredictionFunction<TSrc, TDst> : PredictionFunctionBase<TSrc, TDst>
```

Comment:
>PredictionFunction [](start = 24, length = 18)

I honestly don't understand the purpose of `PredictionFunction`. Why introduce a hierarchy of useless classes? I would much rather rename `PredictionEngine` to `PredictionFunction` and be done with it.

It doesn't have to be done in this PR, but could you just use `PredictionEngine` only, to simplify that future change?

Reply:
I'm with you 100% on this. I have the same thought when I was making this change.

---
In reply to: [236007234](https://github.com/dotnet/machinelearning/pull/1618#discussion_r236007234) [](ancestors = 236007234)